tests/tests/eth_provider.rs

```diff
@@ -819,6 +831,83 @@ async fn test_send_raw_transaction_wrong_signature(#[future] katana: Katana, _se
 
     // Assert that no transaction is found
     assert!(tx.is_none());
+
+    // Verify that the number of pending transactions in the mempool remains unchanged
+    assert_eq!(eth_provider.mempool().unwrap().pool_size().pending, mempool_size.pending);
```

Comment:
Can we also check the transaction hash that is inside the mempool here?

Reply:
Each time also can we check the `total` of the pool size https://github.com/paradigmxyz/reth/blob/019ec727a2e078fd1517d5b04f39f6713d3bf0d7/crates/transaction-pool/src/traits.rs#L1259-L1280 to be sure that everything has been set up correctly in the pending pool and that nothing has been entered by mistake, for example in the blob pool or in the base fee pool

Reply:
Agree, I’ve added the check on the total size of the mempool, while I didn’t add the hash check here because the mempool is empty, as the signature is incorrect. What do you think?